freegs/machine.py

```diff
@@ -442,6 +454,289 @@ def __ne__(self, other):
+        """
+        Updates field attribute of sensor with measured field at that
+        point/direction if sensor is on
+        """
+
+        if self.status:
+
+            field = (tokamak.Br(self.R, self.Z)) * np.cos(self.theta) + (tokamak.Bz(self.R, self.Z)) * np.sin(self.theta)
+
+            if equilibrium is not None:
+                field += (equilibrium.plasmaBr(self.R, self.Z)) * np.cos(self.theta) + (equilibrium.plasmaBz(self.R, self.Z)) * np.sin(self.theta)
+
+            self.measurement = field
+
+    # Methods used in Reconstruction for generating the Greens matrices
+    def get_PlasmaGreensRow(self, equilibrium):
+        return equilibrium.dR * equilibrium.dZ * (GreensBr(equilibrium.R, equilibrium.Z, self.R,self.Z) * np.cos(self.theta) + GreensBz(equilibrium.R,equilibrium.Z, self.R,self.Z) * np.sin(self.theta))
+
+    def get_CoilGreensRow(self, coil):
+        return np.cos(self.theta) * coil.controlBr(self.R,self.Z) + np.sin(self.theta) * coil.controlBz(self.R, self.Z)
+
+    def get_VesselGreensRow(self, passive):
+        return np.cos(self.theta) * passive.controlBr(self.R,self.Z) + np.sin(self.theta) * passive.controlBz(self.R, self.Z)
+
+
+class FluxLoopSensor(Sensor):
+    """
+    Represents position of a Flux Loop Sensor.
+    Consists a single (R,Z) point, describing the position of the probe
+    Contains a method to find the value of flux (psi) at the position
+    """
+    def __init__(self, R, Z, flux=0, name=None, weight=1, status=True, measurement=None):
+        Sensor.__init__(self, R, Z, name=name, weight=weight, status=status, measurement=measurement)
+
+    def get_measure(self, tokamak, equilibrium):
+        """
+            Updates flux attribute with
+            poloidal flux at the point of measurement
+            if sensor is on
+        """
+        if self.status:
+            if equilibrium is not None:
+                psi = equilibrium.psiRZ(self.R,self.Z)
+            else:
+                psi = tokamak.psi(self.R, self.Z)
+
+            self.measurement = psi
+
+    # Methods used in Reconstruction for generating the Greens matrices
+    def get_PlasmaGreensRow(self, equilibrium):
+        return equilibrium.dR * equilibrium.dZ * Greens(equilibrium.R, equilibrium.Z, self.R,self.Z)
+
+    def get_CoilGreensRow(self, coil):
+        return coil.controlPsi(self.R,self.Z)
+
+    def get_VesselGreensRow(self, passive):
+        return passive.controlPsi(self.R,self.Z)
+
+
+class Filament(Coil):
+    """
+    Represents a filament in the vessel structure, capable of carrying currents
+    """
+
+    def __init__(self, R, Z, dR, dZ, resistivity, name, current=0):
+        Coil.__init__(self, R, Z, current=current, control=False)
+        self.name = name
+        self.dR = dR
+        self.dZ = dZ
+        self.resistance = 2 * np.pi * self.R * resistivity / (self.dR * self.dZ)
+
+        self.n_fils = 1
+        self.n_basis = 1
+        self.filaments = [self]
```

Comment:
this is a solution to a formatting problem i had, though not the neatest. I will write some tests to double check there are no issues with doing this. basically i wanted to avoid using ifinstances, so for all elements wihtin the vessel list i have them looping through the each of their elements in the filaments attribute. this means for a single filament, the filaments attribute should be itself, and for a passive it should be all the different filaments contained.

another solution to this would be to make single filament passive objects, and assert all elemtns within the vessel list must be a passive object.